sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -92,67 +94,282 @@ public HttpPipeline getHttpPipeline() {
+                    return Mono.justOrEmpty(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, twinHeaders));
+                } catch (JsonProcessingException e) {
+                    return Mono.error(e);
+                }
+            });
+    }
+
+    /**
+     * Gets a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to retrieve.
+     * @param clazz The model class to convert the relationship to.
+     * @param <T> The generic type to convert the relationship to.
+     * @return The relationship corresponding to the provided relationshipId.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<T> getRelationship(String digitalTwinId, String relationshipId, Class<T> clazz) {
+        return withContext(context -> getRelationship(digitalTwinId, relationshipId, clazz, context))
+            .map(DigitalTwinsResponse::getValue);
+    }
+
+    /**
+     * Gets a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to retrieve.
+     * @param clazz The model class to convert the relationship to.
+     * @param <T> The generic type to convert the relationship to.
+     * @return A REST response containing the relationship corresponding to the provided relationshipId.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<DigitalTwinsResponse<T>> getRelationshipWithResponse(String digitalTwinId, String relationshipId, Class<T> clazz) {
+        return withContext(context -> getRelationship(digitalTwinId, relationshipId, clazz, context));
+    }
+
+    <T> Mono<DigitalTwinsResponse<T>> getRelationship(String digitalTwinId, String relationshipId, Class<T> clazz, Context context) {
+        return protocolLayer
+            .getDigitalTwins()
+            .getRelationshipByIdWithResponseAsync(digitalTwinId, relationshipId, context)
+            .map(response -> {
+                T genericResponse = mapper.convertValue(response.getValue(), clazz);
+                DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                return new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders);
+            });
+    }
+
+    /**
+     * Updates the properties of a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to be updated.
+     * @param relationshipUpdateOperations The application/json-patch+json operations to be performed on the specified digital twin's relationship.
+     * @return An empty response.
```

Comment:
it's not really returning a response

Reply:
True, updated it to "returns an Empty Mono"